internal/fuse/file.go

```diff
@@ -136,9 +138,20 @@ func (f *file) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse.ReadR
 	dst := resp.Data[0:req.Size]
 	readBytes := 0
 	remainingBytes := req.Size
+
+	// The documentation of bazil/fuse actually says that synchronization is
+	// required (see https://godoc.org/bazil.org/fuse#hdr-Service_Methods):
+	//
+	// Multiple goroutines may call service methods simultaneously;
+	// the methods being called are responsible for appropriate synchronization.
+	//
+	// Hence Lock to ensure operation on blobs is not done concurrently
+	// (doesn't work with cached blob in getBlobAt)
+	f.m.Lock()
```

Comment:
defer f.m.Unlock()? There are multiple exits, and lack of memory can cause a panic.

Reply:
is changed.